docs/DeveloperGuide.md

```diff
@@ -1,25 +1,277 @@
 # Developer Guide
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Application Architecture
+![Architecture](images-dg/Architecture.png)
 
+The **Architecture Diagram** given above explains the high-level design of the Flash Card Manager Application.
+
+`Main` is responsible for initializing the other components in the program and linking them up correctly.
+
+* `Ui` Takes in instructions from user and displays the output to the user
+* `Logic` Consists of the `Parsers` and the `Commands`. The `Parser` decipher the user input and executes the specific `Command` that affects the change the user wishes.
+* `Model` Holds the data that is in memory as the program runs. It consists of the 3 components:
+    * `Flash Card` : A single question and answer pair.
+    * `Deck` : A list of `Flash Card`s under a common topic.
+    * `Deck List` : A complete list of all the `Deck`s in memory.
+* `Storage` Reads and writes data from and to a text file.
+
+#### How to **components** interact with one another
+![Sequence Diagram](images-dg/Sequence%20Diagram.png)
+The **Sequence Diagram** above shows how the components interact for a basic `create <deck name>` command where a new deck is created and added in to the `Deck List`.
+
+### User Interface
+
+**API**: [seedu/ecardnomics/Ui.java](https://github.com/AY2021S1-CS2113-T14-2/tp/tree/master/src/main/java/seedu/ecardnomics/Ui.java)
+
+The UI contains String constants that represent the outputs
+that the application is defined to produce.
+
+The `UI` component has two main purposes:
+* Reading user input from the console.
+* Printing program output to the console.
+
+Reading of user input is done using the method `readUserInput()`
+which reads one line of user input. The other methods within `UI` are
+called when a specific output needs to be printed.
+
+The `UI` component passes the user input to the `NormalParser` and
+`DeckParser` components that will extract the relevant information.
+The `UI` component provides its printing methods to `NormalParser`
+and `DeckParser` for printing the appropriate output when required.
+
+### Logic
+#### Overall Logic
+![DG-Overall Logic UML](./images-dg/Logic-DG.png?raw=true "Overall Logic Diagram")
```

Comment:
Should this part be triangular arrow if you means inheritance?
![image](https://user-images.githubusercontent.com/26886705/97536080-960d4280-19f7-11eb-833b-a47962468a16.png)